tests/composite_pk/test_filter.py

```diff
@@ -54,6 +57,20 @@ def test_filter_and_count_user_by_pk(self):
             with self.subTest(lookup=lookup, count=count):
                 self.assertEqual(User.objects.filter(**lookup).count(), count)
 
+    def test_rhs_pk(self):
+        msg = "CompositePrimaryKey cannot be used as a lookup value."
+        with self.assertRaisesMessage(ValueError, msg):
+            Comment.objects.filter(text__gt=F("pk")).count()
+
+    def test_cast_pk(self):
+        qs = Comment.objects.filter(text__gt=Cast(F("pk"), TextField()))
+        if connection.features.can_cast_tuple_to_text:
+            self.assertSequenceEqual(qs, [])
+        else:
+            msg = "This backend doesn't support casting CompositePrimaryKey."
+            with self.assertRaisesMessage(NotSupportedError, msg):
+                qs.count()
```

Comment:
This will need importing but using skipUnlessDBFeature and skipIfDBFeature might be nice here
```suggestion
    @skipUnlessDBFeature("can_cast_tuple_to_text")
    def test_cast_pk_to_text(self):
        qs = Comment.objects.filter(text__gt=Cast(F("pk"), TextField()))
        self.assertSequenceEqual(qs, [])

    @skipIfDBFeature("can_cast_tuple_to_text")
    def test_cannot_cast_pk_to_text(self):
        msg = "This backend doesn't support casting CompositePrimaryKey."
        with self.assertRaisesMessage(NotSupportedError, msg):
            Comment.objects.filter(text__gt=Cast(F("pk"), TextField())).count()
```

Reply:
Might also be worth covering casting it to something else (Json maybe? :shrug:) to cover the postgres when not casted to text